tool/gravity/cli/clusterconfig.go

```diff
@@ -280,7 +281,26 @@ func validateClusterConfig(localEnv *localenv.LocalEnvironment, update libcluste
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	return validate.ClusterConfiguration(existing, update)
+	err = validate.ClusterConfiguration(existing, update)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	server, err := findLocalServer(cluster.ClusterState.Servers)
+	if err != nil {
+		return trace.NotFound("unable to find server info")
+	}
+	if err := validate.NetworkOverlap(server.AdvertiseIP, update.GetGlobalConfig().ServiceCIDR, fmt.Sprintf("The advertise address %v conflicts with the service network CIDR range %v. Please specify a different service CIDR.",
+		server.AdvertiseIP, update.GetGlobalConfig().ServiceCIDR)); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := validate.NetworkOverlap(server.AdvertiseIP, update.GetGlobalConfig().PodCIDR, fmt.Sprintf("The advertise address %v conflicts with the pod network CIDR range %v. Please specify a different pod CIDR.",
```

Comment:
Nit: I would move the message into a separate variable so the lines are shorter and code is easier to read. As a rule of thumb, I use 80-100 characters as a cut-off point for a line length.

```go
message := fmt.Sprintf("xxx", ...)
if err := validate.NetworkOverlap(xxx, yyy, message); err != nil {
```

This applies to all 4 places where this is called.